src/ape/api/providers.py

```diff
@@ -598,6 +598,72 @@ def get_transaction_trace(  # type: ignore[empty-body]
             Iterator(:class:`~ape.type.trace.TraceFrame`): Transaction execution trace.
         """
 
+    @raises_not_implemented
+    def poll_blocks(  # type: ignore[empty-body]
+        self,
+        stop_block: Optional[int] = None,
+        required_confirmations: Optional[int] = None,
+        new_block_timeout: Optional[int] = None,
+    ) -> Iterator[BlockAPI]:  # type: ignore[empty-body]
```

Comment:
only the top type ignore (line 602) is needed i believe

```suggestion
    ) -> Iterator[BlockAPI]: 
```